code/api/src/main/scala/com/avast/datadog4s/api/event/Event.scala

```diff
@@ -0,0 +1,24 @@
+package com.avast.datadog4s.api.event
+
+//TODO add Date support
+/**
+ *
+ * @param title required - The title of the event
+ * @param text required - The text body of the event
+ * @param date required (defaults to the current time from the DogStatsD server) - The epoch timestamp for the event
+ * @param hostname The name of the host
+ * @param aggregationKey A key to use for aggregating events
+ * @param priority Specifies the priority of the event (normal or low).
+ * @param sourceTypeName The source type name
+ * @param alertType error, warning, success, or info (defaults to info)
+ */
+case class Event(
+  title: String,
+  text: String,
+  date: Option[Long] = None,
+  hostname: Option[String] = None,
+  aggregationKey: Option[String] = None,
+  priority: Priority = Priority.Normal,
+  sourceTypeName: Option[String] = None,
+  alertType: AlertType = AlertType.Info
```

Comment:
I think it would be a wroth while to have separate case class for each event. The point of this library is to not have a have a user friendly, higher level API. I think something like 
```
sealed trait Event {
  def title: String
  def text: String
  ...
}

object Event {
  case class Info (...) extends Event
  case class Error (...) extends Event
}
```
is better approach

Reply:
I have implemented this approach but I'm wondering if there is a way to better handle the default values